Gifski/ConversionViewController.swift

```diff
@@ -77,24 +79,39 @@ final class ConversionViewController: NSViewController {
 		timeRemainingEstimator.start()
 
 		progress?.performAsCurrent(withPendingUnitCount: 1) { [weak self] in
-			Gifski.run(conversion) { result in
-				guard let self = self else {
-					return
-				}
-
+			guard let self = self else {
+				return
+			}
+			let gifUrl = self.generateTempGifUrl(for: conversion.video)
+			if let gifData = gifData {
 				do {
-					let gifUrl = self.generateTempGifUrl(for: conversion.video)
-					try result.get().write(to: gifUrl, options: .atomic)
+					try gifData.write(to: gifUrl, options: .atomic)
+
 					try? gifUrl.setMetadata(key: .itemCreator, value: "\(App.name) \(App.version)")
 					defaults[.successfulConversionsCount] += 1
 
 					self.didComplete(conversion: conversion, gifUrl: gifUrl)
 				} catch Gifski.Error.cancelled {
-					self.cancelConversion()
+					cancelConversion()
 				} catch {
-					self.presentError(error, modalFor: self.view.window)
-					self.cancelConversion()
+					presentError(error, modalFor: view.window)
+					cancelConversion()
 				}
+			} else {
+				Gifski.run(conversion, completionHandler: { result in
+					do {
+						try result.get().write(to: gifUrl, options: .atomic)
+					} catch Gifski.Error.cancelled {
+						self.cancelConversion()
+					} catch {
+						self.presentError(error, modalFor: self.view.window)
+						self.cancelConversion()
+					}
+					try? gifUrl.setMetadata(key: .itemCreator, value: "\(App.name) \(App.version)")
```

Comment:
can `gifUrl.setMetadata` && `defaults[. successfulConversionsCount] += 1` be moved to `self.didComplete()`?

Reply:
sure